No Merge: Add additional marker colors and icons

vikasrana1989 opened the pull request.

knownasilya commented on addon/components/google-maps-markup/component.js:
```diff
@@ -260,6 +289,8 @@ export default Ember.Component.extend(ParentMixin, {
             event.stop();
           });
           listeners.pushObjects([ mapListener, dataListener ]);
+        }else if (tool.dataId === 'Point') {
```
Style nitpick

knownasilya commented on addon/components/google-maps-markup/component.js:
```diff
@@ -689,7 +722,24 @@ export default Ember.Component.extend(ParentMixin, {
+            let isDefaultIcon = tool.icon.id === tool.default.icon.id;
+            let isDefaultColor = tool.customColor.id === tool.default.customColor.id;
+
+            if (! isDefaultIcon || ! isDefaultColor) {
```
`if (!isDefaultIcon || !isDefaultColor) {`

knownasilya commented on addon/components/google-maps-markup/template.hbs:
```diff
@@ -38,6 +38,32 @@
+        </div>
+      {{/if}}
+      {{#if (is-equal opt.type 'customColor')}}
+        <div class="dropdown {{if customColorMenu 'open'}}">
```
consistent naming between `iconDropDown` and `customColorMenu`, since both are dropdowns

knownasilya commented on addon/utils/tools.js:
```diff
@@ -19,7 +19,52 @@ export default {
+      id: 'pin',
+      display: 'Pin'
+    }],
+    icon:{
```
spacing

knownasilya replied:
Not sure about putting the defaults and the list items in this config.

knownasilya commented:
Icons http://map-icons.com/

vikasrana1989 commented:
@knownasilya 

We have to add ember drop down component to add the customize drop down. Please check the latest code.
We added the drop down for color as same as it was for others.
We added a new drop down for marker icons.
We tried http://map-icons.com/ but there were some problem in there. We checked and found that[ http://map-icons.com/  is not maintained any more](https://github.com/scottdejonge/map-icons/issues/60#issuecomment-265030528). And owner suggested to use [https://material.io/icons/](https://material.io/icons/) 

Now we are review [https://material.io/icons/](https://material.io/icons/) to update icon color on map.

knownasilya commented on package.json:
```diff
@@ -31,9 +31,11 @@
     "broccoli-asset-rev": "^2.4.5",
     "broccoli-merge-trees": "^1.2.1",
     "broccoli-static-compiler": "^0.2.1",
+    "ember-bootstrap": "1.0.0-alpha.10",
```
I'd rather not use this heavy addon since we don't use it anywhere else in MapGeo, and most likely we'll be updating to material-design or BS4 in the near future.

knownasilya commented:
1. I'd rather not use that addon, maybe just use a select instead of the dropdown.
2. Excellent!
3. Great!
4. See below.
5. https://material.io/icons/ is fine, just add the icons in the map section.

knownasilya commented on addon/components/google-maps-markup/template.hbs:
```diff
@@ -41,18 +41,16 @@
+            {{#each activeTool.icons as |icon|}}
+              <option value={{icon.display}}>
+                <div>
                   <img src={{icon.path}} height="10px" width="10px">
```
You should be able to render the svgs directly without making a file.

knownasilya commented on addon/components/google-maps-markup/template.hbs:
```diff
@@ -41,18 +41,16 @@
-            {{#dd.menu as |ddm|}}
-              {{#each activeTool.icons as |icon|}}
-                <div {{action 'updateOptionValue' activeTool 'icon' icon}}>
+          <select onchange={{action "updateOptionValue" activeTool 'icon' value="target.value"}}>
```
Looking good 👍

knownasilya commented:
Just tried out the latest changes. Things look good. A few things came to mind:

1. The marker dropdown resets to the default after every use of the marker tool
2. The SVG markers could probably be scaled down a bit
3. I'm not sure about having a select for the hover color, we can talk about this in the meeting

knownasilya commented:
Complementary function http://stackoverflow.com/a/37657940/483616

knownasilya commented on addon/utils/tools.js:
```diff
@@ -40,7 +40,11 @@ export default {
+    icon: {
+      id: 'default',
+      display: 'Default',
+      path: 'google-maps-markup/images/spotlight-poi-highlighted_hdpi.png',
```
This icon needs to be removed.

vikasrana1989 commented:
Issues Description:-
1. Marker Icon position is not coming right some time and not properly centered.
2. When we are removing markers, Marker Icons are not getting remove from the map.

Both Issues are fixed and latest code have been updated here. This branch is ready to merge. We are clear now.
